package com.kuit.kuit5.ui.asset.component
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.tooling.preview.Preview
import com.kuit.kuit5.model.AssetData
import androidx.compose.material3.Icon

import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.layout.size
import androidx.compose.ui.unit.dp
import com.kuit.kuit5.R
import com.kuit.kuit5.ui.theme.BankSaladTheme.typography

@Composable
fun AssetInfoItem(modifier: Modifier = Modifier,
                 asset : AssetData

) {
    Row(
        modifier=Modifier
            .fillMaxWidth(),
        verticalAlignment= Alignment.CenterVertically,
        horizontalArrangement= Arrangement.SpaceBetween
    ){
        Row(verticalAlignment=Alignment.CenterVertically){
            Icon(
                modifier= Modifier
                    .size(30.dp),
                painter= painterResource(id= asset.icon),
                contentDescription="asset icon",
                tint = Color.Unspecified
            )
            //Spacer()
            Text(text= asset.name,
                style=typography.body_01_R_14)
        }
        Text(text=asset.value,
        style=typography.body_01_M_14)
    }
}

@Preview(showBackground = true)
@Composable
private fun AsseInfoItemPreview() {
    val asset =AssetData(
        icon = R.drawable.kakao_bank,
        name = "입출금 통장",
        value = "1,000,000원",

    )
    AssetInfoItem(
        asset= asset
    )
}